### DISCUSSION TITLE
How to implement goBack button ?

### DISCUSSION THREAD

**Discussion by arnaudambro** [remix-run/remix#3153] [Q&A] [answered]
Hello there,

How to implement goBack button in Remix ?
`typeof window !== "undefined" ? () => window?.history?.back() : null` ?
Or by importing `history` from `react-router-dom` ?

**Comment by kiliman** [comment, marked as answer]
```ts
import { useNavigate } from '@remix-run/react'

const navigate = useNavigate()
const goBack = () => navigate(-1)

<button onClick={goBack}>Go Back</button>
```
https://reactrouter.com/docs/en/v6/api#usenavigate

**Reply by arnaudambro** [reply]
cool, thanks ! 
I'm a bit confused between `remix` and `react-router` and the docs for each of them
Should we use `react-router` within `remix` or is it already all included ?

Now that you mention `useNavigate` that I didn't know, I discovered this page : https://remix.run/docs/en/v1/other-api/react-router

Everything is clearer now, but I lost quite some time on it ! Thanks for your help

**Reply by kiliman** [reply]
Remix is built on top of React Router. They re-export some components/hooks from that package. The next release of RR will incorporate a lot of Remix enhancements to RR. So Remix will be essentially a server-side compiler for RR.

https://remix.run/blog/remixing-react-router

**Reply by janhesters** [reply]
Late question, but is this button accessible?

I thought navigation's have to be done via `link`? 🤔

**Reply by sergiodxa** [reply]
@janhesters it should be reported as a button, similar to how the native back button will be too anyway.

**Reply by janhesters** [reply]
@sergiodxa Thank you!
